server/imports/core/connection/index.js

```diff
@@ -279,105 +431,318 @@ Connection.prototype = {
+    });
 
     // url
     let connectionUrl = 'mongodb://';
     if (connection.authenticationType) {
       if (username) connectionUrl += encodeURIComponent(username);
-      else if (connection[connection.authenticationType].username) connectionUrl += encodeURIComponent(connection[connection.authenticationType].username);
+      else if (connection[connection.authenticationType].username) {
+        connectionUrl += encodeURIComponent(
+          connection[connection.authenticationType].username
+        );
+      }
 
       if (password) connectionUrl += `:${encodeURIComponent(password)}`;
-      else if (connection[connection.authenticationType].password) connectionUrl += `:${encodeURIComponent(connection[connection.authenticationType].password)}`;
+      else if (connection[connection.authenticationType].password) {
+        connectionUrl += `:${encodeURIComponent(
+          connection[connection.authenticationType].password
+        )}`;
+      }
 
       connectionUrl += '@';
     }
-    connection.servers.forEach((server) => { connectionUrl += `${encodeURIComponent(server.host)}:${server.port},`; });
+    connection.servers.forEach((server) => {
+      connectionUrl += `${encodeURIComponent(server.host)}:${server.port},`;
+    });
 
-    if (connectionUrl.endsWith(',')) connectionUrl = connectionUrl.substring(0, connectionUrl.length - 1);
+    if (connectionUrl.endsWith(',')) {
+      connectionUrl = connectionUrl.substring(0, connectionUrl.length - 1);
+    }
     connectionUrl += '/';
     if (keepDB) connectionUrl += connection.databaseName;
 
     // options
-    if (connection.authenticationType === 'mongodb_cr' || connection.authenticationType === 'scram_sha_1' || connection.authenticationType === 'scram_sha_256') {
-      connectionUrl += ConnectionHelper.addOptionToUrl(connectionUrl, 'authSource', connection[connection.authenticationType].authSource);
+    if (
+      connection.authenticationType === 'mongodb_cr' ||
+      connection.authenticationType === 'scram_sha_1' ||
+      connection.authenticationType === 'scram_sha_256'
+    ) {
+      connectionUrl += ConnectionHelper.addOptionToUrl(
+        connectionUrl,
+        'authSource',
+        connection[connection.authenticationType].authSource
+      );
     } else if (connection.authenticationType === 'mongodb_x509') {
-      connectionUrl += ConnectionHelper.addOptionToUrl(connectionUrl, 'ssl', 'true');
-    } else if (connection.authenticationType === 'gssapi' || connection.authenticationType === 'plain') {
-      if (connection.authenticationType === 'gssapi') connectionUrl += ConnectionHelper.addOptionToUrl(connectionUrl, 'gssapiServiceName', connection.gssapi.serviceName);
-      connectionUrl += ConnectionHelper.addOptionToUrl(connectionUrl, 'authSource', '$external');
+      connectionUrl += ConnectionHelper.addOptionToUrl(
+        connectionUrl,
+        'ssl',
+        'true'
+      );
+    } else if (
+      connection.authenticationType === 'gssapi' ||
```

Comment:
Codacy has a fix for the issue: ['||' should be placed at the beginning of the line.](https://app.codacy.com/gh/nosqlclient/nosqlclient/pullRequest?prid=7730964)
```suggestion
      connection.authenticationType === 'gssapi'
```